signer/src/error.rs

```diff
@@ -3,6 +3,15 @@ pub enum Error {
     #[error("The change amounts for the transaction is negative: {0}")]
     InvalidAmount(i64),
 
+    #[error("Got an old fee estimate")]
+    OldFeeEstimate,
+
+    #[error("Failed to get fee estimates from all fee estimate sources")]
+    NoGoodFeeEstimates,
+
+    #[error("{0}")]
+    Reqwest(#[from] reqwest::Error),
+
```

Comment:
I'm generally in favor of having multiple dedicated error types over a single large type encompassing everything that can ever go wrong across all modules. These variants are very unrelated to the `InvalidAmount` and `Taproot` error variants. So I would prefer to have a dedicated type in `fees.rs` with just these variants, which will make it easier for us to write proper error handling logic in the signer event loop once we start calling these functions.

Reply:
> I'm generally in favor of having multiple dedicated error types over a single large type encompassing everything that can ever go wrong across all modules. 

I probably have a minority view here but I like dedicated error types that encompass everything that can ever go wrong across all modules[^1]. But the dedicated error approach is good, so I'm okay with not enforcing any particular approach here so long as we have enough information to debug when something bad happens.


[^1]: More specifically, I like to follow the approach laid out in [this blog post](https://kazlauskas.me/entries/errors). Basically, the author talks about his experience with one large error enum with enough variants to preserve context, and I think that's the way to go.